File: ts-src/src/cal_circulating_supply.ts
import { getOrbsBalance } from './get_cb_orbs_balance';
import { getTeamBalanceInStaking, getWalletOrbsBalance} from './contract_parser';


/**
 * Calculates the non-circulating supply of ORBS tokens by summing:
 *  - The balance held in Coinbase Prime (portfolio ORBS balance)
 *  - The total supply held by the team in staking (staking contract + staking wallet)
 */

export async function calcNoncirculatingSupply(): Promise<number> {
  const [coinbaseBalance, teamStakingTotal, stakingWalletBalance] = await Promise.all([
    getOrbsBalance(),
    getTeamBalanceInStaking(),
    getWalletOrbsBalance()
  ]);
  const nonCirculating = coinbaseBalance + teamStakingTotal + stakingWalletBalance;
  return nonCirculating;
}

// Allow running this module directly: `ts-node cal_circulating_supply.ts`
if (require.main === module) {
  (async () => {
    try {
      const v = await calcNoncirculatingSupply();
      console.log(v);
    } catch (err) {
      console.error(err);
      process.exit(1);
    }
  })();
}

File: python-src/cal_circulating_supply.py
from get_cb_orbs_balance import main as get_cb_orbs_balance
from contract_parser import main as get_team_balance_in_staking


"""
This script calculates the non-circulating supply of ORBS tokens by summing
the balance held in Coinbase and the total supply held by the team in staking.

"""
def cal_noncirculating_supply():
    coinbase_balance = get_cb_orbs_balance()
    total_supply = get_team_balance_in_staking()
    non_circulating_supply = coinbase_balance + total_supply
    return non_circulating_supply


if __name__ == "__main__":
    print(cal_noncirculating_supply())